### PR TITLE
249 remove analysis variable name plot title in spaghettiplot

### LINKED ISSUE #249: Spahghetti Plot
### Feature description

 - [ ] appears to be the only module where the analysis variable selection is dynamically added to the plot title as well as the plot y-axis label. it is not needed in the title since it is included in the y-axis label.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct.

### Contribution Guidelines

- [X] I agree to follow this project's Contribution Guidelines.

### Security Policy

- [X] I agree to follow this project's Security Policy.

### PR DESCRIPTION
**Pull request by m7pr**
Close #249 

Before the change
title_before

After the change

after

Tested with below

```r
library(stringr)

# original ARM value = dose value
arm_mapping <- list(
  "A: Drug X" = "150mg QD", "B: Placebo" = "Placebo", "C: Combination" = "Combination"
)
color_manual <- c("150mg QD" = "#000000", "Placebo" = "#3498DB", "Combination" = "#E74C3C")

ADLB <- rADLB
var_labels <- lapply(ADLB, function(x) attributes(x)$label)
ADLB <- ADLB %>%
  mutate(AVISITCD = case_when(
    AVISIT == "SCREENING" ~ "SCR",
    AVISIT == "BASELINE" ~ "BL",
    grepl("WEEK", AVISIT) ~
      paste(
        "W",
        trimws(
          substr(
            AVISIT,
            start = 6,
            stop = str_locate(AVISIT, "DAY") - 1
          )
        )
      ),
    TRUE ~ NA_character_
  )) %>%
  mutate(AVISITCDN = case_when(
    AVISITCD == "SCR" ~ -2,
    AVISITCD == "BL" ~ 0,
    grepl("W", AVISITCD) ~ as.numeric(gsub("\\D+", "", AVISITCD)),
    TRUE ~ NA_real_
  )) %>%
  # use ARMCD values to order treatment in visualization legend
  mutate(TRTORD = ifelse(grepl("C", ARMCD), 1,
                         ifelse(grepl("B", ARMCD), 2,
                                ifelse(grepl("A", ARMCD), 3, NA)
                         )
  )) %>%
  mutate(ARM = as.character(arm_mapping[match(ARM, names(arm_mapping))])) %>%
  mutate(ARM = factor(ARM) %>%
           reorder(TRTORD)) %>%
  mutate(ANRLO = .5, ANRHI = 1) %>%
  rowwise() %>%
  group_by(PARAMCD) %>%
  mutate(LBSTRESC = ifelse(USUBJID %in% sample(USUBJID, 1, replace = TRUE),
                           paste("<", round(runif(1, min = .5, max = .7))), LBSTRESC
  )) %>%
  mutate(LBSTRESC = ifelse(USUBJID %in% sample(USUBJID, 1, replace = TRUE),
                           paste(">", round(runif(1, min = .9, max = 1.2))), LBSTRESC
  )) %>%
  ungroup()
attr(ADLB[["ARM"]], "label") <- var_labels[["ARM"]]
attr(ADLB[["ANRLO"]], "label") <- "Analysis Normal Range Lower Limit"
attr(ADLB[["ANRHI"]], "label") <- "Analysis Normal Range Upper Limit"

# add LLOQ and ULOQ variables
ADLB_LOQS <- goshawk:::h_identify_loq_values(ADLB, "LOQFL")
ADLB <- left_join(ADLB, ADLB_LOQS, by = "PARAM")

g_spaghettiplot(
  data = ADLB,
  subj_id = "USUBJID",
  biomarker_var = "PARAMCD",
  biomarker = "CRP",
  value_var = "AVAL",
  trt_group = "ARM",
  time = "AVISITCD",
  color_manual = color_manual,
  color_comb = "#39ff14",
  alpha = .02,
  xtick = c("BL", "W 1", "W 4"),
  xlabel = c("Baseline", "Week 1", "Week 4"),
  rotate_xlab = FALSE,
  group_stats = "median",
  hline_vars = c("ANRHI", "ANRLO"),
  hline_vars_colors = c("pink", "brown")
)

```

### CODE REVIEW

**Comment by m7pr** [comment]
@npaszty the code and the example are related to the removal of the TITLE. the only mistake I made was to put `LEGEND` in the title of this PR/MR but the provided fix removes the unneeded component from the plot TITLE - see print screens from the initial comment

**Review by edelarua — APPROVED** [review]
This looks good to go if Nick is happy with the solution. Thanks @m7pr!

**Comment by npaszty** [comment]
@m7pr 

yes, that's the problem, the printscreen from the initial comment has the incorrect compoonent of the title removed. "AVAL" chould be removed not the not the analysis variable label

File: R/g_spaghettiplot.R [hunk 1]
```diff
@@ -279,7 +279,7 @@ g_spaghettiplot <- function(data,
     select(!!sym(biomarker_var_label)) %>%
     unique() %>%
     magrittr::extract2(1)
-  gtitle <- paste0(biomarker1, unit1, value_var, " Values by Treatment @ Visits")
+  gtitle <- paste0(unit1, value_var, " Values by Treatment @ Visits")
```
Comment (m7pr) [inline R/g_spaghettiplot.R:282]:
```suggestion
  gtitle <- paste0(biomarker1, unit1, " Values by Treatment @ Visits")
```

**Comment by m7pr** [comment]
Ah, got it @npaszty !
Sorry for that. I just send a push with an update.

Removed AVAL and left the biomarker

image

**Pull request merged**